### ISSUE TITLE
DL4J: YOLO - support multiple objects in one location

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#7598]
From gitter:

> Alex Black @AlexDBlack 12:08
@KolyaIvankov yes, it's 1 object per grid cell for each example, by the way the labels are constructed
https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/objdetect/Yolo2OutputLayer.java#L64-L69
As far as predictions: predictions for each bounding box are generated in parallel, so hypothetically you could 2 objects at the same location with different bounding boxes (they can't have the same shape/bounding box)
For training with multiple labels in the same location - it might be possible to support this with some modifications to label format, but I'd need to review the implementation in detail before I could say for sure

**Comment by KolyaIvankov** [comment]
To make it more explicit. Consider the picture as below:
![image](https://user-images.githubusercontent.com/16176232/56567965-4bb33600-65b6-11e9-91fd-699f7106d3eb.png)
According the the present code, the data reader reads the three objects as
```

**Comment by KolyaIvankov** [comment]
One of the possible walkarounds to this problem is contained in YOLOv3. Instead of having just one output layer, YOLOv3 in its current edition has 3 of them, dedicated to large, medium and small objects. Thus, occlusion becomes not a lesser issue, as the occluded object is likely to be picked up by another output layer.

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
